src/Helper/Service/AssetFactory.php

```diff
@@ -22,24 +26,23 @@ class AssetFactory implements FactoryInterface
      */
     public function __invoke(ContainerInterface $container, $name, ?array $options = null)
     {
-        $helper = new Asset();
-
+        /** @psalm-var mixed $config */
         $config = $container->get('config');
-        if (isset($config['view_helper_config']['asset'])) {
-            $configHelper = $config['view_helper_config']['asset'];
-            if (isset($configHelper['resource_map']) && is_array($configHelper['resource_map'])) {
-                $helper->setResourceMap($configHelper['resource_map']);
-            } else {
-                throw new Exception\RuntimeException('Invalid resource map configuration.');
-            }
-        }
+        $config = is_array($config) ? $config : [];
```

Comment:
Should this include `instanceof ArrayAccess` then `iterator_to_array`

Reply:
Not sure about this one: it is very possible that `$config` is an `ArrayAccess`, so IMO we should `assert(is_array($config) || $config instanceof ArrayAccess)` here :thinking:

Reply:
> …is inline with most other components…

Correct.